src/main/java/frc/robot/vision/sources/LimeLightSource.java

```diff
@@ -0,0 +1,163 @@
+package frc.robot.vision.sources;
+
+import edu.wpi.first.math.Pair;
+import edu.wpi.first.math.geometry.Pose3d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Rotation3d;
+import edu.wpi.first.networktables.NetworkTableEntry;
+import edu.wpi.first.networktables.NetworkTableInstance;
+import frc.robot.subsystems.GBSubsystem;
+import frc.robot.vision.VisionConstants;
+import frc.robot.vision.filters.Filter;
+import frc.robot.vision.limelights.LimelightEntryValue;
+import frc.robot.vision.rawdata.RawAprilTagVisionData;
+import frc.utils.Conversions;
+import frc.utils.alerts.Alert;
+import frc.utils.alerts.AlertManager;
+import frc.utils.alerts.PeriodicAlert;
+import frc.utils.time.TimeUtils;
+
+import java.util.Optional;
+
+public class LimeLightSource extends GBSubsystem implements VisionSource<RawAprilTagVisionData> {
+
+	private final NetworkTableEntry robotPoseEntryBotPose2;
+	private final NetworkTableEntry RobotPoseEntryBotPose1;
+	private final NetworkTableEntry aprilTagIdEntry;
+	private final NetworkTableEntry aprilTagPoseEntry;
+	private final NetworkTableEntry robotOrientationEntry;
+	private final String name;
+	private Filter<RawAprilTagVisionData> filter;
+	private double[] robotPoseArray;
+	private double[] aprilTagPoseArray;
+	private Rotation2d robotHeading;
+	private LimelightGyroAngleValues gyroAngleValues;
+	private boolean useOldRobotPoseEntry;
+
+	public LimeLightSource(String name, String parentLogPath, Filter<RawAprilTagVisionData> filter) {
+		super(parentLogPath + name + "/");
+
+		this.name = name;
+		this.filter = filter;
+		this.robotPoseEntryBotPose2 = getLimelightNetworkTableEntry("botpose_orb_wpiblue");
+		this.RobotPoseEntryBotPose1 = getLimelightNetworkTableEntry("botpose_wpiblue");
```

Comment:
```suggestion
		this.robotPoseEntryBotpose2 = getLimelightNetworkTableEntry("botpose_orb_wpiblue");
		this.RobotPoseEntryBotpose1 = getLimelightNetworkTableEntry("botpose_wpiblue");
```

this is a Limelight thing

Reply:
Why do you want botpose2 above botpose1? I don't get it

Reply:
I don't get it either, and I wrote the comment